<launch>
    <node name="gazebo" pkg="gazebo_ros" type="gazebo" required="true" args="$(find simple_youbot_nav)/gazebo/worlds/simple_maze.world"/>
    <node name="sensory_memory" pkg="simple_youbot_nav" type="custom_agent_starter.py" required="true" args="-m SensoryMemory -f $(find simple_youbot_nav)/configs/agent.conf"/>
    <node name="sensory_motor_memory" pkg="simple_youbot_nav" type="custom_agent_starter.py" required="true" args="-m SensoryMotorMemory -f $(find simple_youbot_nav)/configs/agent.conf"/>
    <node name="perceptual_associative_memory" pkg="simple_youbot_nav" type="custom_agent_starter.py" required="true" args="-m PerceptualAssociativeMemory -f $(find simple_youbot_nav)/configs/agent.conf"/>
    <node name="workspace" pkg="simple_youbot_nav" type="custom_agent_starter.py" required="true" args="-m Workspace -f $(find simple_youbot_nav)/configs/agent.conf"/>
    <node name="global_workspace" pkg="simple_youbot_nav" type="custom_agent_starter.py" required="true" args="-m GlobalWorkspace -f $(find simple_youbot_nav)/configs/agent.conf"/>
    <node name="conscious_contents_queue" pkg="simple_youbot_nav" type="custom_agent_starter.py" required="true" args="-m ConsciousContentsQueue -f $(find simple_youbot_nav)/configs/agent.conf"/>
    <node name="procedural_memory" pkg="simple_youbot_nav" type="custom_agent_starter.py" required="true" args="-m ProceduralMemory -f $(find simple_youbot_nav)/configs/agent.conf"/>
    <node name="action_selection" pkg="simple_youbot_nav" type="custom_agent_starter.py" required="true" args="-m ActionSelection -f $(find simple_youbot_nav)/configs/agent.conf"/>
</launch>
